### PR TITLE
Full refactor of QnA page with filters add-on bugFix/401

### PR DESCRIPTION
**Pull request by Elugormo**
## Description

1. For resolving this bug and regarding the UI part, I rebuilt the QnA details page and refactored it by completely deleting it from the source file. To implement the features mentioned in the description of my task, I used some components from Material-UI, specifically, I used the basic concept of accordion-like collapsable fields in a table. To implement the particular feature, I created a completely new component not to interfere with the parent component and basic table styling defined in table-create-like files. So by using my new Collapsable Table View component I implemented the basic collapsable table fields, with accordion-like functionality, where the Admin can write a reply to customers' questions. Also, I did a great job regarding styling and making it look good and marvellous as it possibly could be. 

2. Also, I rewrote hardcoded filters that were created for this component previously and changed them to more basic ones, that are used on several other pages and are default. I completely recreated my FilterByValue because of problematic and incorrect usage. 
3. For the in-system frontend development part, I solved some issues concerning Redux data flow and recreated a new hook to make data fetching easier and smoother. Also, I found several problems regarding pagination and solved them as well. Significant changes were made concerning GraphQl. The hardcoded filter that was built inside the query was falsy, so I did some research not only on the Frontend part but also some changes were applied to the Backend part. Consider typedefs and Filter types that I made for easy fetching, and also I got rid of hardcoded obscuring filters by replacing them with more straightforward and conceivable ones.
4. Basic implementation of the search was implemented using a basic filter navigation component with my own-coded redux actions and GraphQl queries.

#### Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
image  Previous view |  image Updated View
image Old QnA details page | image Updated view with accordion functionality instead of using separate pages
image Old filters | image Updated filters
image Old broken filter | image  New, user-friendly one
image No pagination provided though functionality is essential | image Solved issue

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅ All tests passed locally
- [x] ✨ My changes working with up-to-date front-end and back-end part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

**Review by grebiniuk — CHANGES_REQUESTED** [review]
Please cover your functionality with tests.
There should be at least 80% coverage of the new code.

File: src/containers/table-container-collapsable-row/table-container-collapsable-row.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        className={classes.tableRowCursor}
+        key={id}
+        onClick={() => setOpen(!open)}
+        hover
+      >
+        {showCheckbox && (
+          <TableCell>
+            <Checkbox
+              color='default'
+              inputProps={{ 'aria-label': 'checkbox with default color' }}
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/containers/table-container-collapsable-row/table-container-collapsable-row.js]: fixed

File: src/containers/table-container-collapsable-row/table-container-collapsable-row.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+            )}
+          </TableCell>
+        )}
+      </TableRow>
+      <TableRow className={classes.collapseRowCursor}>
+        <TableCell className={classes.tableCell} colSpan={7}>
+          <Collapse in={open} timeout='auto'>
+            <Box margin={1}>
+              <Typography variant='h5' gutterBottom display='inline'>
+                Запитання від {properties.senderName}:
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/containers/table-container-collapsable-row/table-container-collapsable-row.js]: fixed

File: src/containers/table-container-collapsable-row/table-container-collapsable-row.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+              </Typography>
+              {answer && (
+                <>
+                  <Typography
+                    style={{ margin: '20px 0 5px' }}
+                    variant='h5'
+                    gutterBottom
+                    component='div'
+                  >
+                    Відповідь:
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/containers/table-container-collapsable-row/table-container-collapsable-row.js]: fixed

File: src/containers/table-container-collapsable-row/table-container-collapsable-row.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                {!answer && (
+                  <TextField
+                    id='filled-full-width'
+                    style={{ width: '100%' }}
+                    multiline
+                    rows={6}
+                    margin='normal'
+                    placeholder={labels.emailQuestionsLabels.placeholder}
+                    InputLabelProps={{ shrink: true }}
+                    label='Введіть вашу відповідь'
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/containers/table-container-collapsable-row/table-container-collapsable-row.js]: fixed

File: src/containers/table-container-collapsable-row/table-container-collapsable-row.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                  />
+                )}
+                {!answer && (
+                  <Box>
+                    <Button
+                      variant='contained'
+                      color='primary'
+                      onClick={handleSaveButtonAction}
+                    >
+                      Відповісти
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/containers/table-container-collapsable-row/table-container-collapsable-row.js]: fixed

File: src/pages/email-questions/email-questions-list.js [hunk 6]
```diff
@@ -106,26 +108,31 @@ const EmailQuestionsList = () => {
       ? list.map((question) => {
           const { answer } = question;
 
-          const questionToShow = `<b>Q:</b> ${question.text}`;
+          const questionToShow = `<b>Запитання:</b> ${question.text}`;
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/pages/email-questions/email-questions-list.js]: fixed

File: src/utils/email-question-list.js [hunk 7]
```diff
@@ -1,2 +1,4 @@
 export const answerTextHandler = (answer) =>
-  answer && answer.text ? `<br> <b>A:</b> ${answer.text}` : '';
+  answer && answer.text ? `<br> <b>Відповідь:</b> ${answer.text}` : '';
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/utils/email-question-list.js]: fixed

File: src/utils/questions.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import _ from 'lodash';
+
+export const showQuestionOptions = {
+  PENDING: 'Очікує відповіді',
+  SPAM: 'Спам',
+  ANSWERED: 'Відповідь надано'
+};
+export const placeholderQuestionSearch = 'по тексту';
```
[the remark this answers is not in the document]
Comment (Elugormo) [reply src/utils/questions.js]: fixed

**Review by grebiniuk — APPROVED** [review]

**Review by yuliiakucher — APPROVED** [review]

**Pull request merged**